File: py/fib.py
import sys


def usage():
    print("Usage: python3 fib.py <NUMBER> ")


def fib(num, a=1,b=1):
    if num == 0:
        return a
    elif num == 1:
        return b
    else:
        return fib(num-1, b, a+b)


def main():
    if len(sys.argv) < 2:
        usage()
        raise(Exception("Error: not enough input arguments!"))

    print(fib(int(sys.argv[1])))


if __name__ == "__main__":
    main()

File: pl/fib.pl
use strict;

# std fib calc in recursive impl
sub fib {
    my $num = shift;
    my $a = shift // 0; # defualt 0
    my $b = shift // 1; # default 1
    if ($num == 0) { # case 1
        return $a;
    }
    elsif ($num == 1) { # case 2
        return $b;
    }
    else { # case 3
        return fib($num - 1, $b, $a + $b);
    }
}

# main routine
sub main {
    # grab number from stdin
    my $num = $ARGV[0];

    # assert number given, or error
    if (not defined($num)) {
        die "Error: no input number";
    }

    # calc fib for number
    my $calc = fib $num;

    # print
    print "$calc\n"
}

# injection point
main

